stabilize  tests

MatusKysel opened the pull request:
## Purpose
Fixes: #1588
Fixes: #1568

- sync_five_nodes -> now we are waiting until initial transaction are executed before we continue with the test
- DISABLED_transfer_lot_of_blocks -> @mfrankovi PR should fix that
- node_full_sync -> now not checking for syncing due to empty PBFT blocks but checking non empty blocks
- pbft_manager_run_multi_nodes -> checking for non-empty blocks and increase timeouts because of execution can takes more time 

## How does the solution address the problem
<!-- Describe your solution. -->


## Changes made
<!-- Summary or changes that have been made. -->

JakubFornadel commented on tests/network_test.cpp:
```diff
@@ -80,7 +77,8 @@ TEST_F(NetworkTest, transfer_block) {
 
 // Test creates two Network setup and verifies sending blocks between is successfull
-TEST_F(NetworkTest, transfer_lot_of_blocks) {
+// This test can not work anymore as we are marking other nodes as malicous becasue of invalid dag blocks
```
solution is to send valid blocks and not disabling the test

MatusKysel replied:
@mfrankovi  has a fix in his PR, so this will work for now

JakubFornadel reviewed (changes requested):
I did not check the other changes, but the fix for sync_five_nodes from this PR probably did not do anything -  at least on my PC under Debug it did not help at all. dummy_transaction() that you deleted was not causing any issues and the errors we have for failing txs were not caused by dummy txs. 

At the beginning we sent 5 txs that would transfer some tokens from node0 to all other nodes so they can start sending txs too but for some reason these were not always finalized on all nodes and then when they tried to send tokens, those txs would start failing because of insufficient balance. 

I fixed it here https://github.com/Taraxa-project/taraxa-node/pull/1600

Let's keep the other changes you made and remove the ones from sync_five_nodes test

MatusKysel commented:
> I did not check the other changes, but the fix for sync_five_nodes from this PR probably did not help - at least on my PC under Debug it did not help and I experienced the same errors as before. dummy_transaction() that you deleted was not causing any issues and the errors we have for failing txs were not caused by dummy txs.
> 
> At the beginning we sent 5 txs that would transfer some tokens from node0 to all other nodes so they can start sending txs too but for some reason these were not always finalized on all nodes and then when they tried to send tokens, those txs would start failing because of insufficient balance.
> 
> I fixed it here #1600
> 
> Let's keep the other changes you made and remove the ones from sync_five_nodes test

- you did not fixed the test, just change it's behavior 
- please stop editing my PRs and task assigned to me !!!

gaoqi8649 commented on tests/network_test.cpp:
```diff
@@ -80,7 +77,8 @@ TEST_F(NetworkTest, transfer_block) {
 // Test creates two Network setup and verifies sending blocks between is successfull
-TEST_F(NetworkTest, transfer_lot_of_blocks) {
+// This test can not work anymore as we are marking other nodes as malicous becasue of invalid dag blocks
+TEST_F(NetworkTest, DISABLED_transfer_lot_of_blocks) {
```
Let's enable the transfer_lot_of_blocks unit test, Miho's PR has merged in develop branch

MatusKysel commented:
> 

let's talk about that over the meeting, it will be easier.

MatusKysel commented:
> > > I did not check the other changes, but the fix for sync_five_nodes from this PR probably did not help - at least on my PC under Debug it did not help and I experienced the same errors as before. dummy_transaction() that you deleted was not causing any issues and the errors we have for failing txs were not caused by dummy txs.
> > > At the beginning we sent 5 txs that would transfer some tokens from node0 to all other nodes so they can start sending txs too but for some reason these were not always finalized on all nodes and then when they tried to send tokens, those txs would start failing because of insufficient balance.
> > > I fixed it here #1600
> > > Let's keep the other changes you made and remove the ones from sync_five_nodes test
> > 
> > 
> > 
> > * you did not fixed the test, just change it's behavior
> > * please stop editing my PRs and task assigned to me !!!
> 
> This is such waste of time, we could have had it merged yesterday as both of our solutions fix it, but ok I am going to follow your `"please stop editing my PRs and task assigned to me !!!"` and provide you here a few facts why I edited your PR and unassigned it from you:
> 
>     * when you proposed your first solution (removal of dummy_transaction) I knew for 100% it is not enough so while you were sleeping I debugged it and found out what exactly is the problem - first 5 txs would not be applied on all nodes. [The fix for this is here - this single line](https://github.com/Taraxa-project/taraxa-node/pull/1600/files#diff-cb712a3395b4b23506c8277c7056adf0f3cf679e3e7f04bb3de00f3fb1a68dd4R496). Just this line was enough for the test stop failing (you fixed it differently and achieved the same thing [here](https://github.com/Taraxa-project/taraxa-node/pull/1597/files#diff-cb712a3395b4b23506c8277c7056adf0f3cf679e3e7f04bb3de00f3fb1a68dd4R551))
> 
>     * So the 1 line change was basically enough, but then I realized that the test is not about those first 5 txs anyway(in the next for loop we send approx 50 txs so if there is 50 or 55 it does not really matter) so I simply removed it and made the test little bit faster as we can skip this initial tokens distribution and just put it into the config
> 
> 
> So thats it, I basically did try to help. I fixed the test, edited your PR and and assigned it to me as at that time your solution was not working and I did expect that we simply merge this and move on.

let's talk about that over the meeting, it will be easier.
